### PR TITLE
revamps navigation menus by adding dropdowns

### PR DESCRIPTION
**Pull request by binford2k**
This takes the very busy navigation bars and reorganizes them into
dropdown menus. This gives us a lot more open space, but also lets have
a more intuitive navigation structure.

- moves the "connect with us" links from the footer into a nav menu.
  - also creates a `/connect` page that better explains what's bridged
    and how you can choose protocols to connect with.
- adds a Project Overview page with links to github repo lists
- configures the entire nav structure in `_config.yml` instead of some
  being hardcoded in the html
- adds the ability to add icons to any menu item.
  - if the icons aren't appreciated, just delete from the config
  - updated to current FontAwesome so the online search is usable

I did -not- make the money icon spin around, but I was tempted to.

If you like the overall look & feel, but want to adjust the menu
organization, can I suggest that we merge this and then contribute a
separate PR to reorganize?

This does not yet add any of the OpenPuppet pages. I'll do that in a
separate PR and then add navigation after we decide on a name.

### CODE REVIEW

**Comment by binford2k** [comment]
Standard full width view
Screenshot 2024-12-26 at 3 19 29 PM

Narrow view, like on mobile with the hamburger open
Screenshot 2024-12-26 at 3 20 09 PM

**Comment by binford2k** [comment]
Also, this follows the menu proposal at https://gist.github.com/binford2k/fb71bc23448832d68a4a541af36741bf

**Review by smortex — APPROVED** [review]
Great!

**Review by bastelfreak — APPROVED** [review]

File: _includes/head.html [hunk 1]
```diff
@@ -11,7 +11,11 @@
 
     <link rel="shortcut icon" href="{{site.url}}{{site.baseurl}}/favicon.ico" type="image/x-icon">
     <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC" crossorigin="anonymous">
-    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css">
+
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@6.7.2/css/fontawesome.min.css">
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@6.7.2/css/solid.min.css">
```
Comment (bastelfreak) [inline _includes/head.html:16]: We really need something to automatically update the versions

**Comment by genebean** [comment]
LGTM

**Pull request merged**